Remotion/Core/Extensions/Logging/Log4Net/Log4NetLoggerProvider.cs

```diff
@@ -0,0 +1,110 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Reflection;
+using log4net.Appender;
+using log4net.Config;
+using log4net.Core;
+using log4net.Layout;
+using log4net.Repository.Hierarchy;
+using Microsoft.Extensions.Logging;
+using Remotion.Reflection;
+using Remotion.Utilities;
+
+namespace Remotion.Logging.Log4Net;
+
+public class Log4NetLoggerProvider : ILoggerProvider
+{
+  public void Initialize ()
+  {
+    XmlConfigurator.Configure();
+  }
+
+  public void InitializeConsole ()
+  {
+    var appender = CreateConsoleAppender();
+    BasicConfigurator.Configure(appender);
+  }
+
+  public void InitializeConsole (Microsoft.Extensions.Logging.LogLevel defaultThreshold, params LogThreshold[] logThresholds)
```

Comment:
This might actually not be required anymore. @MichaelKetting? 
Only DependDB usecases are re-motion itself.

If useless now, we can remove the convert method in `Log4NetLogger` in this task.

Reply:
All three Initialize-Methods:

originally they where instance to allow interface implementations. This is now an obsolete concept. If an application developer wants to use log4net, they can use the underlying APIs so we can drop all the Initialize-methods on this type, leaving only the interface-declared methods as public api.

@Frazerus 
Please remind me, did I already mention that we also want to keep a copy of the original APIs in Remotion.Obsolete to look them up via DependDB? it mihgt be in one of the later tasks.

Reply:
You did not mention that before concerning the initialize things. For (almost) everything else we should have the APIs still in the extension methods to provide backwards compatibility.

Reply:
@Frazerus this sounds about right. The reasoning is:
1. only stuff without a log4net dependency can get moved into an extension method in Remotion.Extensions.
2. those methods could indeed get moved to an extension method in Remtoion.Logging.Log4Net, class `Log4NetLoggerProviderExtensions` and marked as obsolete. I went with the default of deleting based on recommendation plus my own thought on "there's not good place to put them" in combination with almost no usage at all. If you feel it's better to move them as obsolete to extension method, I'm okay with that, too.